Tests iOS/ReportARecommendationTests.swift

```diff
@@ -0,0 +1,100 @@
+import XCTest
+import Sails
+
+
+class ReportARecommendationTests: XCTestCase {
+    var server: Application!
+    var app: PocketAppElement!
+    
+    override func setUpWithError() throws {
+        continueAfterFailure = false
+        
+        let uiApp = XCUIApplication()
+        app = PocketAppElement(app: uiApp)
+        
+        server = Application()
+        
+        server.routes.post("/graphql") { _, _ in
+            Response {
+                Status.ok
+                Fixture.load(name: "slates")
+                    .replacing("PARTICLE_JSON", withFixtureNamed: "particle-sample", escape: .encodeJSON)
+                    .data
+            }
+        }
+        
+        server.routes.post("/com.snowplowanalytics.snowplow/tp2") { _, _ in
+            return Response {
+                Status.ok
+                Data()
+            }
+        }
+        
+        try server.start()
+        
+        let arguments = (
+            PocketAppElement.defaultLaunchArguments
+            + [
+                "clearKeychain",
+                "clearCoreData",
+                "clearImageCache"
+            ]
+        ).filter { $0 != "disableSnowplow" }
+        
+        let environment = PocketAppElement.defaultEnvironment.merging(
+            [
+                "accessToken": "test-access-token",
+                "sessionGUID": "session-guid",
+                "sessionUserID": "session-user-id",
+            ]
+        ) { _, new in new }
+        app.strictLaunch(
+            arguments: arguments,
+            environment: environment
+        )
```

Comment:
I feel like we can come up with a better abstraction here. What if we had like a `LaunchArguments` struct. It could have a property for each argument with defaults for each property. Then you could just initialize the struct and specify the properties you want to change.

Something like this:
```swift
struct PocketAppElement {
  init(args: LaunchArguments) {
    // map each property of launchArguments to an array
  )
}

struct LaunchArgument {
  let clearKeychain: Bool
  let clearCoreData: Bool
  let clearImageCache: Bool
  let disableSentry: Bool
  let disableSnowplow: Bool

  init(
    clearKeychain: Bool = true, 
    clearCoreData: Bool = true, 
    clearImageCache: Bool = true, 
    disableSentry: Bool = true,
    disableSnowplow: Bool = true
}

// then at the call site you can specify only the arg you care about
// and allow the rest to use defaults
let args = LaunchArguments(disableSnowplow: false)
app.launch(args: args)
```

Thinking we could do something similar for the launch env vars as well